### ISSUE TITLE
Error in Kibana: no permissions for indices:data/read/msearch 

### ISSUE THREAD

**Issue by sqtran** [openshift/origin-aggregated-logging#645]
I'm seeing the following error messages in Kibana.

`_Error: [security_exception] no permissions for indices:data/read/msearch
    at respond (https://kibana-logging.192.168.42.115.nip.io/bundles/commons.bundle.js?v=10229:96868:16)
    at checkRespForFailure (https://kibana-logging.192.168.42.115.nip.io/bundles/commons.bundle.js?v=10229:96831:8)
    at https://kibana-logging.192.168.42.115.nip.io/bundles/commons.bundle.js?v=10229:95449:8
    at processQueue (https://kibana-logging.192.168.42.115.nip.io/bundles/commons.bundle.js?v=10229:42452:29)
    at https://kibana-logging.192.168.42.115.nip.io/bundles/commons.bundle.js?v=10229:42468:28
    at Scope.$eval (https://kibana-logging.192.168.42.115.nip.io/bundles/commons.bundle.js?v=10229:43696:29)
    at Scope.$digest (https://kibana-logging.192.168.42.115.nip.io/bundles/commons.bundle.js?v=10229:43507:32)
    at Scope.$apply (https://kibana-logging.192.168.42.115.nip.io/bundles/commons.bundle.js?v=10229:43804:25)
    at done (https://kibana-logging.192.168.42.115.nip.io/bundles/commons.bundle.js?v=10229:38253:48)
    at completeRequest (https://kibana-logging.192.168.42.115.nip.io/bundles/commons.bundle.js?v=10229:38451:8)_`

If I enable DEBUG in the logging-fluentd daemonset, I see these messages in /var/log/fluentd.log

_`2017-09-07 18:38:35 +0000 [warn]: dump an error event: error_class=NoMethodError error="undefined method `utc' for \"2017-09-07T18:38:35.135195952Z\":String" tag="kubernetes.var.log.containers.docker-registry-1-w9w53_default_registry-fd1b828787af447ad15897f64f58854452114bc6eddcba35e57a548bda2e662c.log" time=1504809515 record={"log"=>"172.17.0.1 - - [07/Sep/2017:18:38:35 +0000] \"GET /healthz HTTP/1.1\" 200 0 \"\" \"Go-http-client/1.1\"\n", "stream"=>"stdout", "time"=>"2017-09-07T18:38:35.135195952Z", "docker"=>{"container_id"=>"fd1b828787af447ad15897f64f58854452114bc6eddcba35e57a548bda2e662c"}, "kubernetes"=>{"container_name"=>"registry", "namespace_name"=>"default", "pod_name"=>"docker-registry-1-w9w53", "pod_id"=>"48530b12-84da-11e7-8603-722db22c57b2", "labels"=>{"deployment"=>"docker-registry-1", "deploymentconfig"=>"docker-registry", "docker-registry"=>"default"}, "host"=>"localhost", "master_url"=>"https://kubernetes.default.svc.cluster.local", "namespace_id"=>"27b6788a-84da-11e7-8603-722db22c57b2"}, "message"=>"172.17.0.1 - - [07/Sep/2017:18:38:35 +0000] \"GET /healthz HTTP/1.1\" 200 0 \"\" \"Go-http-client/1.1\"\n", "level"=>"info", "hostname"=>"localhost"}
2017-09-07 18:38:35 +0000 [warn]: dump an error event: error_class=NoMethodError error="undefined method `utc' for \"2017-09-07T18:38:35.135216459Z\":String" tag="kubernetes.var.log.containers.docker-registry-1-w9w53_default_registry-fd1b828787af447ad15897f64f58854452114bc6eddcba35e57a548bda2e662c.log" time=1504809515 record={"log"=>"172.17.0.1 - - [07/Sep/2017:18:38:35 +0000] \"GET /healthz HTTP/1.1\" 200 0 \"\" \"Go-http-client/1.1\"\n", "stream"=>"stdout", "time"=>"2017-09-07T18:38:35.135216459Z", "docker"=>{"container_id"=>"fd1b828787af447ad15897f64f58854452114bc6eddcba35e57a548bda2e662c"}, "kubernetes"=>{"container_name"=>"registry", "namespace_name"=>"default", "pod_name"=>"docker-registry-1-w9w53", "pod_id"=>"48530b12-84da-11e7-8603-722db22c57b2", "labels"=>{"deployment"=>"docker-registry-1", "deploymentconfig"=>"docker-registry", "docker-registry"=>"default"}, "host"=>"localhost", "master_url"=>"https://kubernetes.default.svc.cluster.local", "namespace_id"=>"27b6788a-84da-11e7-8603-722db22c57b2"}, "message"=>"172.17.0.1 - - [07/Sep/2017:18:38:35 +0000] \"GET /healthz HTTP/1.1\" 200 0 \"\" \"Go-http-client/1.1\"\n", "level"=>"info", "hostname"=>"localhost"}
2017-09-07 18:38:45 +0000 [warn]: dump an error event: error_class=NoMethodError error="undefined method `utc' for \"2017-09-07T18:38:45.13498464Z\":String" tag="kubernetes.var.log.containers.docker-registry-1-w9w53_default_registry-fd1b828787af447ad15897f64f58854452114bc6eddcba35e57a548bda2e662c.log" time=1504809525 record={"log"=>"172.17.0.1 - - [07/Sep/2017:18:38:45 +0000] \"GET /healthz HTTP/1.1\" 200 0 \"\" \"Go-http-client/1.1\"\n", "stream"=>"stdout", "time"=>"2017-09-07T18:38:45.13498464Z", "docker"=>{"container_id"=>"fd1b828787af447ad15897f64f58854452114bc6eddcba35e57a548bda2e662c"}, "kubernetes"=>{"container_name"=>"registry", "namespace_name"=>"default", "pod_name"=>"docker-registry-1-w9w53", "pod_id"=>"48530b12-84da-11e7-8603-722db22c57b2", "labels"=>{"deployment"=>"docker-registry-1", "deploymentconfig"=>"docker-registry", "docker-registry"=>"default"}, "host"=>"localhost", "master_url"=>"https://kubernetes.default.svc.cluster.local", "namespace_id"=>"27b6788a-84da-11e7-8603-722db22c57b2"}, "message"=>"172.17.0.1 - - [07/Sep/2017:18:38:45 +0000] \"GET /healthz HTTP/1.1\" 200 0 \"\" \"Go-http-client/1.1\"\n", "level"=>"info", "hostname"=>"localhost"}
2017-09-07 18:38:45 +0000 [warn]: dump an error event: error_class=NoMethodError error="undefined method `utc' for \"2017-09-07T18:38:45.135042459Z\":String" tag="kubernetes.var.log.containers.docker-registry-1-w9w53_default_registry-fd1b828787af447ad15897f64f58854452114bc6eddcba35e57a548bda2e662c.log" time=1504809525 record={"log"=>"172.17.0.1 - - [07/Sep/2017:18:38:45 +0000] \"GET /healthz HTTP/1.1\" 200 0 \"\" \"Go-http-client/1.1\"\n", "stream"=>"stdout", "time"=>"2017-09-07T18:38:45.135042459Z", "docker"=>{"container_id"=>"fd1b828787af447ad15897f64f58854452114bc6eddcba35e57a548bda2e662c"}, "kubernetes"=>{"container_name"=>"registry", "namespace_name"=>"default", "pod_name"=>"docker-registry-1-w9w53", "pod_id"=>"48530b12-84da-11e7-8603-722db22c57b2", "labels"=>{"deployment"=>"docker-registry-1", "deploymentconfig"=>"docker-registry", "docker-registry"=>"default"}, "host"=>"localhost", "master_url"=>"https://kubernetes.default.svc.cluster.local", "namespace_id"=>"27b6788a-84da-11e7-8603-722db22c57b2"}, "message"=>"172.17.0.1 - - [07/Sep/2017:18:38:45 +0000] \"GET /healthz HTTP/1.1\" 200 0 \"\" \"Go-http-client/1.1\"\n", "level"=>"info", "hostname"=>"localhost"}
`_

Are these messages normal?  I'm wondering if something is broken and isn't sending logs to elasticsearch, which is giving kibana an error because the logs are empty.  What should I check next?

**Comment by jcantrill** [comment]
Note the logging stack is made of multiple components:
1. Fluentd - The collector of log messages
2. Elasticsearch - The storage for log messages
3. Kibana - Visualization of log messages.

The errors here are unrelated. 

In regards to your Kibana security issue, that will most likely be fixed by either [1] or [2] depending upon what version you are testing and if you are in operations user.

Please consider making available the output of [3] so we may properly diagnose the fluentd issue.

[1] https://github.com/openshift/origin-aggregated-logging/pull/637
[2] https://github.com/openshift/origin-aggregated-logging/pull/571
[3] https://github.com/openshift/origin-aggregated-logging/blob/master/hack/logging-dump.sh

**Comment by richm** [comment]
also please provide the output of
```
oc exec $fluentd_pod -- gem list
```
and - are you using docker with the json-file driver?

**Comment by sqtran** [comment]
@jcantrill Output of logging-dump.sh attached
@richm Output of gem list included in the attached zip.  I'm using Minishift  v1.4.1+0f658ea, which I think is using the default json-file driver.

[logging-20170908_101030.zip](https://github.com/openshift/origin-aggregated-logging/files/1288139/logging-20170908_101030.zip)

**Comment by richm** [comment]
fluentd daemonset has this:
```
        image: sqtran/origin-logging-fluentd:steve23
```
What changes have you made to the Dockerfile and other config?

Also, this gem:
```
fluent-plugin-viaq_data_model (0.0.7)
```
please use the recently released 0.0.8 instead.

**Comment by sqtran** [comment]
Seems like the issue went away after I did a minishift delete and rm -rf ~/.minishift.
